atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/impl/expectations/TrackingClosableIteratorTest.java

```diff
@@ -17,85 +17,100 @@
 package com.palantir.atlasdb.transaction.impl.expectations;
 
 import static org.assertj.core.api.Assertions.assertThat;
-import static org.mockito.Mockito.spy;
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
 
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterators;
 import com.palantir.common.base.ClosableIterator;
 import com.palantir.common.base.ClosableIterators;
-import java.util.ArrayList;
-import java.util.Iterator;
 import java.util.List;
-import java.util.function.Consumer;
 import java.util.function.ToLongFunction;
-import one.util.streamex.StreamEx;
 import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
 
+@RunWith(MockitoJUnitRunner.class)
 public final class TrackingClosableIteratorTest {
     private static final String STRING = "test";
-    private static final ClosableIterator<String> ONE_ELEMENT_ITERATOR =
-            ClosableIterators.wrapWithEmptyClose(List.of(STRING).iterator());
-    private static final ImmutableList<String> STRINGS = ImmutableList.of(
-            "test4", "test4", "test200", "composite", "", "t", "twentyElementString1", "tt", "twentyElementString2");
-
-    // these have to be anonymous inner classes rather than lambdas in order to spy
-    private static final Consumer<Long> NO_OP = new Consumer<>() {
-        @Override
-        public void accept(Long _unused) {}
-    };
-    private static final ToLongFunction<String> STRING_LENGTH_MEASURER = new ToLongFunction<>() {
-        @Override
-        public long applyAsLong(String value) {
-            return value.length();
-        }
-    };
+
+    @Mock
+    private BytesReadTracker tracker;
+
+    @Mock
+    private ToLongFunction<String> measurer;
 
     @Test
-    public void oneElementTrackingClosableIteratorIsWiredCorrectly() {
-        Consumer<Long> tracker = spy(NO_OP);
-        ToLongFunction<String> measurer = spy(STRING_LENGTH_MEASURER);
-        TrackingClosableIterator<String> trackingIterator =
-                new TrackingClosableIterator<>(ONE_ELEMENT_ITERATOR, tracker, measurer);
+    public void oneElementTrackingIteratorIsWiredCorrectly() {
+        long measuredValue = 1L;
+        when(measurer.applyAsLong(anyString())).thenReturn(measuredValue);
```

Comment:
nit: Since we know what the string is (STRING), we should be specific and just have `applyAsLong(STRING)`